src/components/Problems/Problems.tsx

```diff
@@ -58,6 +57,63 @@ interface Series {
   semester: number
 }
 
+const Problem: FC<{
+  problem: Problem
+  registered: boolean
+  commentCount: number
+  setDisplaySideContent: Dispatch<
+    SetStateAction<{
+      type: string
+      problemId: number
+      problemNumber: number
+    }>
+  >
+  canRegister: boolean
+}> = ({problem, registered, commentCount, setDisplaySideContent, canRegister}) => {
+  const handleDiscussionButtonClick = () => {
+    setDisplaySideContent((prevState) => {
+      if (prevState.type === 'discussion' && prevState.problemId === problem.id) {
+        return {type: '', problemId: -1, problemNumber: -1}
+      } else {
+        return {type: 'discussion', problemId: problem.id, problemNumber: problem.order}
+      }
+    })
+  }
+  const handleUploadClick = () => {
+    if (canRegister) {
+      setDisplaySideContent((prevState) => {
+        if (prevState.type === 'uploadProblemForm' && prevState.problemId === problem.id) {
+          return {type: '', problemId: -1, problemNumber: -1}
+        } else {
+          return {type: 'uploadProblemForm', problemId: problem.id, problemNumber: problem.order}
+        }
+      })
+    } else {
+      // ToDo handle if not register
+    }
+  }
+  return (
+    <div className={styles.problem}>
+      <h3 className={styles.problemTitle}>{problem.order}. ÚLOHA</h3>
+      <Latex>{problem.text}</Latex>
+      <div className={styles.actions}>
+        {registered ? <Link href={`/api/competition/problem/${problem.id}/my-solution/`}>moje riešenie</Link> : <></>}
+        {registered ? (
+          <Link href={`/api/competition/problem/${problem.id}/corrected-solution/`}>
+            opravené riešenie ({problem.submitted?.score || '?'})
+          </Link>
+        ) : (
+          <></>
+        )}

```

Comment:
toto sa da napisat aj takto - jednak spojit obe do jedneho fragmentu (`<>xxx</>`), nejak miesto prazdneho fragmentu pouzit `null`:
```ts
{registered ? (
  <>
    <Link href={`/api/competition/problem/${problem.id}/my-solution/`}>moje riešenie</Link>
    <Link href={`/api/competition/problem/${problem.id}/corrected-solution/`}>
      opravené riešenie ({problem.submitted?.score || '?'})
    </Link>
  </>
) : null
)}
```
kazdopadne, mozno chceme spravit to iste co navrhujem pri "odovzdat" - miesto schovavania buttonov ich ukazat a disablenut